app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+            } else if (dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+                await searchDynamicOptions(text);
+            }
+
+            setLoading(false);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [dataSource, term, searchResults]);
+
+    const filterSearchData = (source: string, searchData: DataType, searchTerm: string) => {
+        if (!data) {
+            return [];
+        }
+
+        const lowerCasedTerm = searchTerm.toLowerCase();
+
+        if (source === ViewConstants.DATA_SOURCE_USERS) {
+            return filterProfilesMatchingTerm(searchData as UserProfile[], lowerCasedTerm);
+        } else if (source === ViewConstants.DATA_SOURCE_CHANNELS) {
+            return filterChannelsMatchingTerm(searchData as Channel[], lowerCasedTerm);
+        } else if (source === ViewConstants.DATA_SOURCE_DYNAMIC) {
+            return searchData;
+        }
+
+        return (searchData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().startsWith(lowerCasedTerm));
+    };
+
+    const getMultiselectData = useCallback((): Selection => {
+        let myItems;
+        let multiselectItems: Selection = [];
+
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                myItems = multiselectSelected as Dictionary<UserProfile>;
+                multiselectItems = multiselectItems as UserProfile[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                myItems = multiselectSelected as Dictionary<Channel>;
+                multiselectItems = multiselectItems as Channel[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            default:
+                myItems = multiselectSelected as Dictionary<DialogOption>;
+                multiselectItems = multiselectItems as DialogOption[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+        }
+    }, [multiselectSelected]);
+
+    // Effects
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, onHandleMultiselectSubmit, [onHandleMultiselectSubmit]);
+
+    useEffect(() => {
+        setLoading(true);
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            getChannels();
+        } else {
+            getDynamicOptionsLocally();
+        }
+
+        setLoading(false);
```

Comment:
This doesn't make any sense.

You set the loading to true, call a few asynchronous functions (which will be sent to the queue, you are not awaiting them), and then set loading back to false. Which in practice (unless the setstates get queued (which will be even worse)) just keeps the loading to false.

Reply:
I understand the problem, not sure what's the best solution though, do you have any ideas?

Reply:
`setLoading(false)` should be called eventually (in case of error or finish) by the getter (i.e. `getDynamicOptionsLocally`, etc...)